extensions/cornerstone-dicom-sr/src/getSopClassHandlerModule.ts

```diff
@@ -205,69 +204,100 @@ async function _load(displaySet, servicesManager: AppTypes.ServicesManager, exte
   });
 }
 
+/**
+ * Checks if measurements can be added to a display set.
+ *
+ * @param srDisplaySet - The source display set containing measurements.
+ * @param newDisplaySet - The new display set to check if measurements can be added.
+ * @param dataSource - The data source used to retrieve image IDs.
+ * @param servicesManager - The services manager.
+ */
 function _checkIfCanAddMeasurementsToDisplaySet(
   srDisplaySet,
   newDisplaySet,
   dataSource,
   servicesManager: AppTypes.ServicesManager
 ) {
   const { customizationService } = servicesManager.services;
+
   let unloadedMeasurements = srDisplaySet.measurements.filter(
     measurement => measurement.loaded === false
   );
 
+  /** All measurements were loaded */
   if (unloadedMeasurements.length === 0) {
-    // All already loaded!
     return;
   }
 
-  if ((!newDisplaySet) instanceof ImageSet) {
-    // This also filters out _this_ displaySet, as it is not an ImageSet.
+  if (!(newDisplaySet instanceof ImageSet)) {
     return;
   }
 
   if (newDisplaySet.unsupported) {
     return;
   }
 
-  const { sopClassUids } = newDisplaySet;
+  const { sopClassUids, images } = newDisplaySet;
 
-  // Check if any have the newDisplaySet is the correct SOPClass.
+  /** Check correct SOPClassUID */
   unloadedMeasurements = unloadedMeasurements.filter(measurement =>
-    measurement.coords.some(coord =>
-      sopClassUids.includes(coord.ReferencedSOPSequence.ReferencedSOPClassUID)
-    )
+    measurement.coords.some(coord => {
+      for (let i = 0; i < images.length; ++i) {
```

Comment:
This is a fairly long loop going over both the measurements and coordinates.  Is there a way to only do it for coordinates containing a frame of reference, and then figure out which image inside a volume should apply using the distance calculations from the volume display set?

Reply:
Updated!